ptype/inference.py

```diff
@@ -0,0 +1,260 @@
+    Args:
+        ds (xr.dataset): Dataset of pressure level variables
+        varsP (list): List of pressure level variables to flatten
+        vertical_level_name (str): Name of the pressure level dimension
+    Returns:
+         Pandas Dataframe of flattened variables split out by variable name in format: <varName>_<pressure_level>
+    """
+    l, cols = [], []
+    for v in varsP:
+        for p in ds[vertical_level_name].values:
+            cols.append(f"{v}_{int(p)}")
+            l.append(ds[v].sel(isobaricInhPa=p).values.reshape(-1, 1))
+    flat_data = np.concatenate(l, axis=1)
+
+    return pd.DataFrame(flat_data, columns=cols)
+
+
+def kelvin_to_celcius(temp):
+    """ Convert Kelvin Temperatures to Celcius."""
+    return temp - 273.15
+
+
+def download_data(date, model, product, save_dir, forecast_range):
+    """ Download data use Herbie for specified dates, model and forecast range.
+    Args:
+        date (List of pandas date times): List of Model initialization times
+        model (str): Model to download data from. Supports "hrrr", "rap", or "nam"
+        product (str): Product string for NWP grib data Herbie Accessor
+        save_dir (str): Directory to save model data to
+        forecast_range (tuple): Tuple of (start forecast hour, end_forecast_hour)
+    Returns:
+        None
+    """
+    forecast_hours = range(forecast_range["start"],
+                           forecast_range["end"] + forecast_range["interval"],
+                           forecast_range["interval"])
+    h = FastHerbie(DATES=[date],
+                   model=model,
+                   product=product,
+                   save_dir=save_dir,
+                   fxx=forecast_hours)
+    h.download()
+    return
+
+
+def load_data(var_dict, file, model, drop):
+    """
+    Load variables from grib file, flatten pressure variables and convert to DataFrame. Supports "gfs", "rap", "hrrr"
+    and "nam" models.
+    Args:
+        var_dict: Dictionary of variables to process. Requires "isobaricInPa", "surface", and "heightAboveGround"
+        file: Path to grib file.
+        model: Model name. Supports "gfs", "rap", "hrrr" and "nam".
+        drop: Whether to drop pressure level variables for final written output (not dropped from flattened df).
+
+    Returns:
+        xarray dataset, flatten pandas dataframe, surface data (flattened df)
+    """
+    grib_data = []
+    for key, value in var_dict.items():
+        if key == "product":
+            continue
+        for var in value:
+            grib = cfgrib.open_dataset(file, backend_kwargs={
+                "filter_by_keys": {'typeOfLevel': key, 'cfVarName': var, 'stepType': 'instant'}})
+            if len(grib) == 0:
+                grib = cfgrib.open_dataset(file, backend_kwargs={
+                    "filter_by_keys": {'typeOfLevel': key, 'shortName': var, 'stepType': 'instant'}})
+            grib_data.append(grib)
+
+    ds = xr.merge(grib_data, compat='override')#.load()
```

Comment:
Can you use more robust variable names to describe the different parts of the pipeline. ds and df are really generic, so it can be hard to track their functionality.